backend/remote-state/s3/client.go

```diff
@@ -28,7 +35,58 @@ type RemoteClient struct {
 	lockTable            string
 }
 
-func (c *RemoteClient) Get() (*remote.Payload, error) {
+var (
+	// The amount of time we will retry a state waiting for it to match the
+	// expected checksum.
+	consistencyRetryTimeout = 10 * time.Second
+
+	// delay when polling the state
+	consistencyRetryPollInterval = 2 * time.Second
+
+	// checksum didn't match the remote state
+	errBadChecksum = errors.New("invalid state checksum")
```

Comment:
I was thinking this was too far removed from the ui to provide good feedback, but It looks like this will end up being put directly into a ui.Error.

I'm only checking the error value in one place right now too, so I think the UX trumps the clean value comparison in this case. I'll add your error message string.